src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -10631,7 +10632,8 @@ instruct xorI_rReg_imm(rRegI dst, immI src, rFlagsReg cr)
 
 instruct xorI_rReg_rReg_imm_ndd(rRegI dst, rRegI src1, immI src2, rFlagsReg cr)
 %{
-  predicate(UseAPX);
+  // Strict predicate check to make selection of xorI_rReg_im1_ndd cost agnostic if immI src2 is -1.
+  predicate(UseAPX && n->in(2)->bottom_type()->is_int()->get_con() != -1);
```

Comment:
This strict predicate check could be removed.

Reply:
It's over a general immI match and make selection more robust and agnostic to static patten cost.